rosetta/views.py

```diff
@@ -1,46 +1,262 @@
-import django
+import hashlib
+import json
+import os
+import os.path
+import re
+try:
+    # Python 3
+    from urllib.parse import urlencode
+except ImportError:
+    # Python 2
+    from urllib import urlencode
+import zipfile
+
 from django.conf import settings
-from django.contrib.auth.decorators import user_passes_test
+from django.contrib import messages
+from django.contrib.auth.decorators import login_required, user_passes_test
 from django.core.paginator import Paginator
-if django.VERSION < (1, 10):  # NOQA
-    from django.core.urlresolvers import reverse  # NOQA
-else:  # NOQA
-    from django.urls import reverse  # NOQA
-
 from django.http import Http404, HttpResponseRedirect, HttpResponse
-from django.shortcuts import render
-from django.utils.encoding import iri_to_uri
-from django.utils.translation import ugettext_lazy as _
 from django.views.decorators.cache import never_cache
-from django.utils.encoding import force_text
-from django.contrib import messages
-
+from django.views.generic import TemplateView, View
+try:
+    from django.urls import reverse
+except ImportError:
+    from django.core.urlresolvers import reverse
+from django.utils.decorators import method_decorator
+from django.utils.functional import cached_property
+from django.utils.translation import ugettext_lazy as _
 from microsofttranslator import Translator, TranslateApiException
+from polib import pofile
+import six
+import unicodedata
 
+from rosetta import get_version as get_rosetta_version
+from rosetta.access import can_translate, can_translate_language
 from rosetta.conf import settings as rosetta_settings
-from polib import pofile
 from rosetta.poutil import find_pos, pagination_range, timestamp_with_timezone
 from rosetta.signals import entry_changed, post_save
 from rosetta.storage import get_storage
-from rosetta.access import can_translate, can_translate_language
 
-import json
-import re
-import rosetta
-import unicodedata
-import hashlib
-import os
-import six
 
+def get_app_name(path):
+    return path.split('/locale')[0].split('/')[-1]
```

Comment:
IIUC this requires that local dirs must be a subdir of the django app, but there are projects where app and local are siblings (and django is OK).

Reply:
This function predates this particular pull request, but if it's a problem IRL, it probably warrants being filed as its own Issue.